setupbase.py

```diff
@@ -375,6 +375,15 @@ def npm_components(self):
                 shutil.rmtree(bower_pkg)
             shutil.copytree(npm_pkg, bower_pkg)
 
+    def patch_codemirror(self):
+        """Patch CodeMirror until https://github.com/codemirror/CodeMirror/issues/4454 is resolved"""
+        
+        try:
+            run(['cp', 'scripts/codemirror.js', 'notebook/static/components/codemirror/lib'], cwd=repo_root)
```

Comment:
Rather that `run(['cp'`, it might be better to use [`shutil.copy`](https://docs.python.org/3/library/shutil.html#shutil.copy), which is probably safer.